mscorlib/system/exception.cs

```diff
@@ -385,7 +407,32 @@ public virtual String HelpLink
                 _helpURL = value;
             }
         }
-    
+
+#if MONO
+        public virtual string Source {
+            get {
+                if (_source == null) {
+                    StackTrace st = new StackTrace (this, true);
+                    if (st.FrameCount > 0) {
```

Comment:
OK, I am going to leave the TargetSite change as is though since that has support methods.